bash2048.sh

```diff
@@ -262,6 +264,76 @@ function reload_game {
   index_max=board_size-1
 }
 
+# print the content of the highscores file
+function show_highscores {
+    header="Highscore Name Date"
+    printf "%10s | %30s | %30s" "Highscore" "Name" "Date"
+    echo ""
+    echo "----------------------------------------------------------------------------"
+    while IFS=";" read -r score name date
+    do
+        printf "%10s | %30s | %30s" "$score" "$name" "$date"
+        echo ""
+    done < "${highscore_file}"
+
+}
+
+# check if a score is a high score (in top 10 highest scores)
+# input:
+#         $1: score value
+# output:
+#         0 if true
+#         1 if false
+function is_high_score {
+    if [[ -f "$highscore_file" ]]; then
+        least_score=$(cat $highscore_file | cut -f 1 -d';' | sort -rn| tail -n 1)
+        if [[ "${1}" -ge "${least_score}" ]]; then
+            return 0
+        fi
+    fi
+    return 1
+}
+
+# save a new highscore
+# inputs:
+#         $1: score value
+#         $2: score timestamp
+function save_highscore {
+    # highscore file doesn't exist or is not readable
+    if [[ ! -f "${highscore_file}" ]] || [[ ! -r "${highscore_file}" ]]; then
+        touch "${highscore_file}"
+    fi
+    echo ""
+    read -p "Please enter your name: " name
+    new_highscore="${1} ${name} $(date -d @${2})"
```

Comment:
here we are saving highscore with just spaces between fields, but earlier we expected this file to be delimited by semicolons